erpnext/stock/report/product_bundle_balance/product_bundle_balance.py

```diff
@@ -0,0 +1,187 @@
+		select item.name as item_code, item.item_name, pb.description, item.item_group, item.brand, item.stock_uom
+		from `tabItem` item
+		inner join `tabProduct Bundle` pb on pb.new_item_code = item.name
+		where ifnull(item.disabled, 0) = 0 {0}
+	""".format(conditions), filters, as_dict=1)  # nosec
+
+	parent_items = []
+	for d in parent_item_details:
+		parent_items.append(d.item_code)
+		item_details[d.item_code] = d
+
+	if parent_items:
+		child_item_details = frappe.db.sql("""
+			select
+				pb.new_item_code as parent_item, pbi.item_code, item.item_name, pbi.description, item.item_group, item.brand,
+				item.stock_uom, pbi.uom, pbi.qty
+			from `tabProduct Bundle Item` pbi
+			inner join `tabProduct Bundle` pb on pb.name = pbi.parent
+			inner join `tabItem` item on item.name = pbi.item_code
+			where pb.new_item_code in ({0})
+		""".format(", ".join(["%s"] * len(parent_items))), parent_items, as_dict=1)  # nosec
+	else:
+		child_item_details = []
+
+	child_items = set()
+	for d in child_item_details:
+		if d.item_code != d.parent_item:
+			pb_details.setdefault(d.parent_item, []).append(d)
+			child_items.add(d.item_code)
+			item_details[d.item_code] = d
+
+	child_items = list(child_items)
+	return item_details, pb_details, parent_items, child_items
+
+
+def get_stock_balance(filters, items):
+	sle = get_stock_ledger_entries(filters, items)
+	stock_balance = frappe._dict()
+	for d in sle:
+		stock_balance.setdefault(d.item_code, frappe._dict())[d.warehouse] = d
+	return stock_balance
+
+
+def get_stock_ledger_entries(filters, items):
+	if not items:
+		return []
+
+	item_conditions_sql = ' and sle.item_code in ({})' \
+		.format(', '.join(['"' + frappe.db.escape(i, percent=False) + '"' for i in items]))
+
+	conditions = get_sle_conditions(filters)
+
+	return frappe.db.sql("""
+		select
+			sle.item_code, sle.warehouse, sle.qty_after_transaction, sle.company
+		from
+			`tabStock Ledger Entry` sle force index (posting_sort_index)
+		left join `tabStock Ledger Entry` sle2 on
+			sle.item_code = sle2.item_code and sle.warehouse = sle2.warehouse
+			and (sle.posting_date, sle.posting_time, sle.name) < (sle2.posting_date, sle2.posting_time, sle2.name)
+		where sle2.name is null and sle.docstatus < 2 %s %s""" % (item_conditions_sql, conditions), as_dict=1)  # nosec
+
+
+def get_parent_item_conditions(filters):
+	conditions = []
+
+	if filters.get("item_code"):
+		conditions.append("item.item_code = %(item_code)s")
+	else:
+		if filters.get("brand"):
+			conditions.append("item.brand=%(brand)s")
+		if filters.get("item_group"):
+			conditions.append(get_item_group_condition(filters.get("item_group")))
+
+	conditions = " and ".join(conditions)
+	return "and {0}".format(conditions) if conditions else ""
+
+
+def get_sle_conditions(filters):
+	conditions = ""
+	if not filters.get("date"):
+		frappe.throw(_("'Date' is required"))
+
+	conditions += " and sle.posting_date <= '%s'" % frappe.db.escape(filters.get("date"))
```

Comment:
No need to put ```%s``` inside quotes if you are escaping
I am getting this error
![image](https://user-images.githubusercontent.com/42651287/58561971-f79b1180-8245-11e9-98ac-60b86249234e.png)
```suggestion
	conditions += " and sle.posting_date <= %s" % frappe.db.escape(filters.get("date"))
```